DirectX11/HackerDXGI.cpp

```diff
@@ -167,6 +167,104 @@ HackerDXGISwapChain::HackerDXGISwapChain(IDXGISwapChain *pSwapChain, HackerDevic
 	}
 }
 
+HackerDXGISwapChain::HackerDXGISwapChain(IDXGISwapChain *pSwapChain, HackerDevice *pDevice, HackerContext *pContext, const DXGI_SWAP_CHAIN_DESC* fake_swap_chain_desc, UINT new_width, UINT new_height)
```

Comment:
Cool, sounds good.  For the HackerDXGISwapChain members, you won't need them to be virtual, because typically they already have an implementation.  You can just use the same routine signature and it will be overridden.  It looks like the key one will be the destructor.

(Even stuff I did not want to override needed to be overridden because of the funky Com object approach.  So I kinda-sorta convert all the Com objects we use into actual C++ objects.)

In CreateSwapChain, you'll create the proper version directly, or fall back to standard object, based on d3dx.ini requirements.

<br>
If possible, the best syntax will be to use the follow-on constructor style I show there.  <br>
"... : HackerDXGISwapChain(pSwapChain, pDevice, pContext)".  <br><br>

If something can't be rearranged or it doesn't make sense, you can always call the superclass directly with HackerDXGISwapChain::XX.